app/ui/src/app/api/providers/api-http-provider.service.ts

```diff
@@ -165,9 +165,40 @@ export class ApiHttpProviderService extends ApiHttpService {
   }
 
   private replaceNameMatches(endpointTemplate: string, paramsMap: StringMap<any>): string {
-    return endpointTemplate.replace(/\{(\D*?)\}/g, (fullMatch, ...matchGroups) => {
-      return paramsMap[matchGroups[0].trim()];
+    const matched = [];
+    const answer = endpointTemplate.replace(/\{(\D*?)\}/g, (fullMatch, ...matchGroups) => {
+      const match = matchGroups[0].trim();
+      matched.push(match);
+      return paramsMap[match];
     });
+    // Turn any remaining parameters into query params
```

Comment:
We could probably use https://angular.io/api/common/http/HttpParams to make sure we properly encode the query string.

Reply:
yeah, good call, will add that, thanks!

Reply:
Perhaps you will want to use the `HttpUrlEncodingCodec` class instead (it's not injectable, you must instantiate it with `new`). It features both `encodeKey()` and `encodeValue()` methods.